cmake/CMakeLists.txt

```diff
@@ -2157,65 +1365,109 @@ if (onnxruntime_USE_WINML)
-  include(onnxruntime_python.cmake)
+  message(STATUS "Python Build is enabled")
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_python)
 endif()
 
 if (onnxruntime_BUILD_OBJC)
   message(STATUS "Objective-C Build is enabled")
-  include(onnxruntime_objectivec.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_objectivec)
 endif()
 
 if (onnxruntime_BUILD_UNIT_TESTS)
-  include(onnxruntime_unittests.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_unittests)
+endif()
+
+if (onnxruntime_BUILD_WINML_TESTS)
+  list(APPEND ONNXRUNTIME_TARGETS winml_unittests)
 endif()
 
+# onnxruntime_training depends on onnxruntime_unittests since onnxruntime_training.cmake uses a variable `TEST_SRC_DIR`
+# that is defined in onnxruntime_unittests.cmake
 if (onnxruntime_ENABLE_TRAINING)
-  include(onnxruntime_training.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_training)
   if (onnxruntime_ENABLE_TRAINING_E2E_TESTS)
-    include(onnxruntime_training_e2e_tests.cmake)
+    list(APPEND ONNXRUNTIME_TARGETS onnxruntime_training_e2e_tests)
   endif()
 endif()
 
 if (onnxruntime_BUILD_CSHARP)
   message(STATUS "CSharp Build is enabled")
 #  set_property(GLOBAL PROPERTY VS_DOTNET_TARGET_FRAMEWORK_VERSION "netstandard2.0")
-  include(onnxruntime_csharp.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_csharp)
 endif()
 
 if (onnxruntime_BUILD_WEBASSEMBLY)
   message(STATUS "WebAssembly Build is enabled")
-  include(onnxruntime_webassembly.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_webassembly)
 endif()
 
 if(onnxruntime_BUILD_KERNEL_EXPLORER)
   message(STATUS "Kernel Explorer Build is enabled")
-  include(onnxruntime_kernel_explorer.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_kernel_explorer)
+endif()
+
+if (onnxruntime_ENABLE_EAGER_MODE)
+  if (NOT onnxruntime_ENABLE_TRAINING OR NOT onnxruntime_ENABLE_PYTHON)
+    message(
+        FATAL_ERROR
+        "Option onnxruntime_ENABLE_EAGER_MODE can only be used when onnxruntime_ENABLE_TRAINING and onnxruntime_ENABLE_PYTHON are enabled")
+  endif()
+  add_compile_definitions(ENABLE_EAGER_MODE)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_eager)
+endif()
+
+if (onnxruntime_ENABLE_LAZY_TENSOR)
+  # To support LazyTensor, ORT needs to call Python function from C/C++.
+  # so onnxruntime_ENABLE_PYTHON is required.
+  if (NOT onnxruntime_ENABLE_TRAINING OR NOT onnxruntime_ENABLE_PYTHON)
+    message(
+        FATAL_ERROR
+        "Option onnxruntime_ENABLE_LAZY_TENSOR can only be set when onnxruntime_ENABLE_TRAINING and onnxruntime_ENABLE_PYTHON are enabled")
+  endif()
+  add_compile_definitions(ENABLE_LAZY_TENSOR)
+  # TODO: In the future, we can compile LazyTensor into a standalone
+  # library target, onnxruntime_lazy_tensor, to make the buid
+  # cleaner.
+  #list(APPEND ONNXRUNTIME_TARGETS onnxruntime_lazy_tensor)
 endif()
 
+# When GDK_PLATFORM is set then WINAPI_FAMILY is defined in gdk_toolchain.cmake (along with other relevant flags/definitions).
+if (WIN32 AND NOT GDK_PLATFORM)
```

Comment:
No, we can't. We can't let it change how we build protoc.exe and etc, because it won't work.  This link option is for Windows OneCore and is just for us.